src/test/java/com/parasoft/demoapp/graphql/OrderGraphQLDataFetcherTest.java

```diff
@@ -279,17 +286,233 @@ public void test_createOrder_noPermission() throws IOException {
         assertError_createOrder(response, HttpStatus.FORBIDDEN, ConfigMessages.USER_HAS_NO_PERMISSION);
     }
 
+    @Test
+    public void test_updateOrderByOrderNumber_normal_purchaser() throws Throwable {
+        OrderEntity order = getOrderWithExpectedStatus(OrderStatus.APPROVED, "comment", false, true );
+        OrderStatusDTO orderStatusDTO = new OrderStatusDTO(order.getStatus(), "changed", true, false);
+        ObjectNode variables = objectMapper.createObjectNode();
+        variables.put("orderNumber", order.getOrderNumber());
+        variables.putPOJO("orderStatusDTO", orderStatusDTO);
+
+        GraphQLResponse response = graphQLTestTemplate
+                .withBasicAuth(purchaser.getUsername(), purchaser.getPassword())
+                .perform(UPDATE_ORDER_BY_ORDER_NUMBER_GRAPHQL_RESOURCE, variables);
+
+        assertThat(response).isNotNull();
+        assertThat(response.isOk()).isTrue();
+        response.assertThatNoErrorsArePresent()
+                .assertThatDataField().isNotNull()
+                .and()
+                .assertThatField(UPDATE_ORDER_BY_ORDER_NUMBER_DATA_JSON_PATH)
+                .as(OrderEntity.class)
+                .matches((result) ->
+                        result.getStatus() == order.getStatus() &&
+                                result.getComments().equals(order.getComments()) &&
+                                result.getReviewedByPRCH() &&
+                                result.getReviewedByAPV()
+                );
+    }
+
+    @Test
+    public void test_updateOrderByOrderNumber_normal_approver() throws Throwable {
+        OrderEntity order = getOrderWithExpectedStatus(OrderStatus.PROCESSED, null, true, true );
```

Comment:
Do we need to reset database for this test case? If not, please merge the pull request.